autometacal/python/tf_ngmix/pixels.py

```diff
@@ -0,0 +1,148 @@
+"""
+Observation implementations from ngmix ported to tensorflow
+
+Author: esheldon et al. (original), andrevitorelli (port)
+
+ver: 0.0.0
+
+"""
+
+
+import tensorflow as tf
+import numpy as np
+
+###jacobians
+def make_diagonal_jacobian(row0, col0, scale):
+  """make a simple diagonal jacobian, defining centre & pixel scale"""
+  
+  jacob={'row0': row0,
+  'col0': col0,
+  'dvdrow': scale,
+  'dvdcol': 0.0,
+  'dudrow': 0.0,
+  'dudcol': scale,
+  'det': scale*scale,
+  'scale': scale}
+
+  return jacob
+
+def jacobian_get_vu(jacob, row, col):
+  """
+  convert row,col to v,u using the input jacobian
+  """
+
+  rowdiff = row - jacob['row0']
+  coldiff = col - jacob['col0']
+
+  v = jacob['dvdrow']*rowdiff + jacob['dvdcol']*coldiff
+  u = jacob['dudrow']*rowdiff + jacob['dudcol']*coldiff
+
+  return v, u
+
+def jacobian_get_area(jacob):
+  """
+  get the pixel area
+  """
+  
+  return jacob['scale']**2
+
+#####make "observation" - in our case, it's just the pixels. 
+
+def make_pixels(image, weight, jacob, ignore_zero_weight=True):
```

Comment:
I don't think this will work :-/ We want all our images to be natively TensorFlow tensors, we can't go through a stage where they are turned into a numpy array of pixels. 
You can't do autometacal if you need to go through this "make_pixels" function

Reply:
But maybe I'm wrong, do you have an example of autometacal with this function?

Reply:
You're right. My first thought was that we could change all images to pixel-tensors before using them, but since metacal operations take images, etc, that doesn't work with our idea for a wrapper using tf_ngmix. Later this afternoon I've started the change of make_pixels using tf.meshgrid, so this will be fully tf.